### PR TITLE
Super duper warnings

### PR DESCRIPTION
**Pull request by janisozaur**
15 months it's been broken (since https://github.com/OpenRCT2/OpenRCT2/pull/8641) and only a handful of things to fix, I'm proud of you guys.

### CODE REVIEW

**Comment by janisozaur** [comment]
hey @Gymnasiast can you run this on your machine and report back?

**Comment by Gymnasiast** [comment]
@janisozaur You mean you want the error output when compiling?

**Comment by janisozaur** [comment]
@Gymnasiast I'm fixing the errors spotted by CIs (I did expect some…), please wait a bit

**Comment by Gymnasiast** [comment]
500a4e9ebb6a7305715a202ea3f4fc4f851328c1 (Address possible nullptr dereferences) compiles fine on my machine.

**Comment by janisozaur** [comment]
@Gymnasiast the CI is done. Can you check now?

File: src/openrct2/management/Research.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ uint8_t gResearchExpectedDay;
 std::optional<ResearchItem> gResearchNextItem;
 
 // 0x01358844[500]
-ResearchItem gResearchItems[MAX_RESEARCH_ITEMS];
+// ResearchItem gResearchItems[MAX_RESEARCH_ITEMS];
```
Comment (Gymnasiast) [inline src/openrct2/management/Research.cpp:55]: This line can be removed altogether. It has been replaced with the two vectors below it.

**Comment by Gymnasiast** [comment]
`34ff3319b892edeb3efa62cde42ebd6b657b3b0f` compiles for me without any issue.

**Pull request merged**